SDL/StateInfo.cpp

```diff
@@ -498,9 +498,11 @@ void SDL::Variable::copy(const SDL::Variable& rhs) {
         break;
     case e_VarCreatable:
         for (size_t i = 0; i < minsize; ++i) {
-            rhs.m_data->m_creatable[i]->ref();
-            m_data->m_creatable[i]->unref();
-            m_data->m_creatable[i] = rhs.m_data->m_creatable[i];
+            MOUL::Creatable::SafeUnref(m_data->m_creatable[i]);
+            if (rhs.m_data->m_creatable[i]) {
+                rhs.m_data->m_creatable[i]->ref();
+                m_data->m_creatable[i] = rhs.m_data->m_creatable[i];
+            }
```

Comment:
We need to keep the `ref()` call before the `SafeUnref`, in case the pointers refer to the same object.  I'd also suggest keeping the assignment unconditional, since SafeUnref doesn't set the pointer to `nullptr`.

Reply:
Good points. When I was writing SafeUnref, it did actually set the pointer to `nullptr`, of course that made it not play nicely with destructors. Guess we can't have everything in life...